src/robot/libraries/BuiltIn.py

```diff
@@ -2232,7 +2245,15 @@ def wait_until_keyword_succeeds(self, retry, retry_interval, name, *args):
                     raise AssertionError("Keyword '%s' failed after retrying "
                                          "%s. The last error was: %s"
                                          % (name, message, err))
-                self._sleep_in_parts(retry_interval)
+                keyword_runtime = time.time() - execution_time
+                if strict_interval:
+                    wait_time = retry_interval - keyword_runtime
+                if wait_time >= 0.0:
+                    self._sleep_in_parts(wait_time)
+                else:
+                    logger.warn("Interval violation: retry_interval is {:.3}s"
+                                ", but keyword runtime is {:.3}s."
+                                .format(retry_interval, keyword_runtime))
```

Comment:
This warning is logged only when using the strict mode and keyword execution took more time than the retry interval, right? Few things to be enhanced related to that:

1. The logic would be easier to understand if the code would be structured something like this:
   ```python
   if strict_interval:
       wait_time = retry_interval - keyword_runtime
       if wait_time < 0:
           logger.warn(...)
       self._sleep_in_parts(wait_time)
   ```

2. Actual times could be logged in more human readable format by using `secs_to_timestr` utility method. I'm not sure is the actual keyword run time interesting, because you can see that easily from the log file also otherwise. Something more generic like this could be enough:
   ```python
   logger.warn("Interval violation: Keyword execution time was longer than "
               "retry interval %s.' % secs_to_timestr(retry_interval))

3. There needs to be a test validating the warning. Log message validation can be done on the `atest/robot` side using `Check Log Message` like some other tests in the same suite are doing. If you want to log the actual execution time, you need to use patterns in the expected log message because that time cannot be controlled exactly.